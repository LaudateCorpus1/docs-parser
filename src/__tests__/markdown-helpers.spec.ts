import * as fs from 'fs';
import * as path from 'path';
import * as MarkdownIt from 'markdown-it';

import { safelyJoinTokens } from '../markdown-helpers';

describe('markdown-helpers', () => {
  describe('safelyJoinTokens', () => {
    it('should join no tokens to an empty string', () => {
      expect(safelyJoinTokens([])).toBe('');
    });

    describe('snapshots', () => {
      const fixtureDir = path.resolve(__dirname, 'fixtures');
      for (const markdownFixture of fs.readdirSync(fixtureDir)) {
        if (!markdownFixture.endsWith('.md')) continue;

        it(`should be correct for ${path.basename(markdownFixture, '.md')}`, () => {
          const markdown = new MarkdownIt();
          const tokens = markdown.parse(
            fs.readFileSync(path.resolve(fixtureDir, markdownFixture), 'utf8'),
            {},
          );
          expect(safelyJoinTokens(tokens)).toMatchSnapshot();
        });
      }
    });
  });
});
