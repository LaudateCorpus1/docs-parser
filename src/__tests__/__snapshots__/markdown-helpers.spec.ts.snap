// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for basic-paragraph 1`] = `"This is just a basic paragraph.  It has multiple sentences and natural soft breaks."`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for blockquotes 1`] = `
"This is a paragraph

> and here is a quote

and another paragraph"
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for list-after-paragraph 1`] = `
"This is a paragraph

* this
* is
* a
* list"
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for lists 1`] = `
"* bar
* thing
* stuff
* my tab key does not work and I am sad"
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for multiple-paragraphs 1`] = `
"This is paragraph 1, it has a few sentences.  Like this one, and a soft break.

And now this is a second paragraph, and it's also quite short with a soft break."
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for nested-list 1`] = `
"* top level
  * second level
* back to top
  * back to second
  * still at second
    * now at third"
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for paragraph-fence-removal 1`] = `
"This fence should be removed

Hey"
`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for paragraph-with-inline-code 1`] = `"This is a \`inline code\` block that is \`code\` tagged."`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for paragraph-with-links 1`] = `"This paragraph has a bunch of stuff.  Also some links Foo, these links should be stripped."`;

exports[`markdown-helpers safelyJoinTokens snapshots should be correct for paragraph-with-text-markers 1`] = `"Heya, this is **bold** and _italic_ and *italic* and ~~struck through~~"`;
